go/vt/sqlparser/random_expr.go

```diff
@@ -67,20 +77,28 @@ func (g *generator) atMaxDepth() bool {
 Note: It's important to update this method so that it produces all expressions that need precedence checking.
 It's currently missing function calls and string operators
 */
-func (g *generator) expression() Expr {
+func (g *Generator) Expression() (Expr, string) {
+	typ := "tinyint"
 	if g.randomBool() {
-		return g.booleanExpr()
+		return g.booleanExpr(), typ
 	}
 	options := []exprF{
 		func() Expr { return g.intExpr() },
 		func() Expr { return g.stringExpr() },
 		func() Expr { return g.booleanExpr() },
 	}
 
-	return g.randomOf(options)
+	fn := g.randomOf(options)
+	if fn == g.intExpr() {
```

Comment:
this code seems fragile. I suggest you change the `typ` variable inside the `options` functions instead